### DISCUSSION TITLE
Managing a form with multiple actions

### DISCUSSION THREAD

**Discussion by Frantss** [react-hook-form/react-hook-form#10230] [Q&A]
Hi, I have a form where the user can trigger multiple actions on the same form, is there a community preferred way of doing so? 

Like saving a entity as a draft or in it's entirety, something like this:
<img width="325" alt="Screenshot 2023-04-05 at 12 08 09" src="https://user-images.githubusercontent.com/26557483/230123734-1d763fd8-c541-4a2d-8ee7-8976695dcedb.png">

I want to handle different form validation for each and `form.formState.isSubmitting` states

**Comment by sega057** [comment]
Hello!
That's how I implemented it with zod validation and it looks good to me:

```
export const draftResolver = zodResolver<zod.ZodType<Partial<YourPayloadType>>>(
    zod.object({
        id: zod.string({
            required_error: 'id is required',
        }),
    }),
);

export const defaultResolver = zodResolver<zod.ZodType<YourPayloadType>>(
...

// Inside a component
const isDraft = React.useRef(false);

const methods = useForm({
    resolver: (...args) => {
        if (isDraft.current) {
            return draftResolver(...args);
        }
        return defaultResolver(...args);
    },
    defaultValues,
});

const onSubmit = (draft?: boolean) => {
    isDraft.current = Boolean(draft);

    methods.handleSubmit(async (data) => {
        // any logic you want to submit the form
    })();
};
```
